soc/arm/st_stm32/stm32h7/mpu_regions.c

```diff
@@ -21,18 +23,36 @@ static const struct arm_mpu_region mpu_regions[] = {
 	MPU_REGION_ENTRY("SRAM3_ETH_DESC",
 					 DT_REG_ADDR(DT_NODELABEL(sram3)),
 					 REGION_PPB_ATTR(REGION_256B)),
-#else
+#else /* DT_NODE_HAS_STATUS(DT_NODELABEL(sram3), okay) */
 	MPU_REGION_ENTRY("SRAM2_ETH_BUF",
 					 DT_REG_ADDR(DT_NODELABEL(sram2)),
 					 REGION_RAM_NOCACHE_ATTR(REGION_16K)),
 	MPU_REGION_ENTRY("SRAM2_ETH_DESC",
 					 DT_REG_ADDR(DT_NODELABEL(sram2)),
 					 REGION_PPB_ATTR(REGION_256B)),
-#endif
-#endif
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(sram3), okay) */
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(mac), okay) */
 
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(dma1), okay) || \
+	DT_NODE_HAS_STATUS(DT_NODELABEL(dma2), okay)
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(sram3), okay)
+	MPU_REGION_ENTRY("SRAM3_DMA_BUFS",
+					 (uint32_t)&_sdma_stm32,
+					 REGION_RAM_NOCACHE_ATTR(REGION_4K)),
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(sram2), okay)
+	MPU_REGION_ENTRY("SRAM2_DMA_BUFS",
+					 (uint32_t)&_sdma_stm32,
+					 REGION_RAM_NOCACHE_ATTR(REGION_4K)),
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(sram1), okay)
+	MPU_REGION_ENTRY("SRAM1_DMA_BUFS",
+					 (uint32_t)&_sdma_stm32,
+					 REGION_RAM_NOCACHE_ATTR(REGION_4K)),
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(sram3), okay) */
 	/* DT-defined regions */
 	LINKER_DT_REGION_MPU(ARM_MPU_REGION_INIT)
```

Comment:
This macro should always be called, it can't be conditional.

Reply:
No longer applicable as a different implementation has been requested and applied.